File: testdata/bench/csharp/BenchmarkTest00020.cs
using Microsoft.AspNetCore.Mvc;
using System.Data.SqlClient;

namespace BenchApp.Controllers
{
    public class SqliSafeController : Controller
    {
        [HttpGet]
        public IActionResult Login()
        {
            string user = Request.Query["username"];
            string pass = Request.Query["password"];
            var conn = new SqlConnection("Server=localhost;Database=app;");
            conn.Open();
            var cmd = new SqlCommand("SELECT * FROM users WHERE username = @user AND password = @pass", conn);
            cmd.Parameters.AddWithValue("@user", user);
            cmd.Parameters.AddWithValue("@pass", pass);
            var reader = cmd.ExecuteReader();
            return Ok();
        }
    }
}

File: testdata/bench/lua/BenchmarkTest00020.lua
local pgmoon = require "pgmoon"
local args = ngx.req.get_post_args()
local user = args.username
local pass = args.password
local pg = pgmoon.new({host="127.0.0.1", database="auth"})
pg:connect()
local res = pg:query("SELECT * FROM accounts WHERE username = $1 AND password = $2", user, pass)
ngx.say(res)
